contracts/ConsensusUtils.sol

```diff
@@ -206,6 +218,24 @@ contract ConsensusUtils is EternalStorage, ValidatorSet {
     uintStorage[CURRENT_CYCLE_END_BLOCK] = block.number + getCycleDurationBlocks();
   }
 
+  function _checkJail(address[] _validatorSet) internal {
+    uint256 expectedNumberOfBlocks = ((CYCLE_DURATION_BLOCKS / _validatorSet.length) * VALIDATOR_PRODUCTIVITY)/10;
```

Comment:
can you explain how does this work?
1. I see that blockcounter is increased in cycle(address), is this called every block by ALL validators?
2. isnt a validator supposed to have blocks relative to his stake?
3. what is the magic number 7 of VALIDATOR_PRODUCTIVITY ? the update said less than 90% will be jailed has this seems more like 70%?

Reply:
@leonprou

Reply:
Hey @sirpy 

1. Cycle is called by the validator who mines the block. 
2. No each validator mines the same number of blocks in round robin regardless of stake (stake effects the block emission)
3. Yes this will be update to 9 this is just a work in progress PR (I need to push some changes here)

Reply:
VALIDATOR_PRODUCTIVITY

Reply:
Thanks for checking this out @sirpy